src/commands/add.rs

```diff
@@ -1,53 +1,337 @@
 use anyhow::bail;
 use is_terminal::IsTerminal;
-use std::collections::HashMap;
-use strum::IntoEnumIterator;
+use std::{
+    collections::{BTreeMap, HashMap},
+    time::Duration,
+};
+use strum::{Display, EnumIs, EnumIter, IntoEnumIterator};
 
 use crate::{
+    consts::TICK_STRING,
     controllers::{database::DatabaseType, project::ensure_project_and_environment_exist},
-    util::prompt::prompt_multi_options,
+    util::prompt::{
+        fake_select, prompt_multi_options, prompt_options, prompt_text,
+        prompt_text_with_placeholder_disappear, prompt_text_with_placeholder_if_blank,
+    },
 };
 
 use super::*;
 
-/// Provision a database into your project
+/// Add a service to your project
 #[derive(Parser)]
 pub struct Args {
     /// The name of the database to add
     #[arg(short, long, value_enum)]
     database: Vec<DatabaseType>,
+
+    /// The name of the service to create (leave blank for randomly generated)
+    #[clap(short, long)]
+    service: Option<Option<String>>,
```

Comment:
Whats up with the nested options?

Reply:
To allow for --service to be specified without a value, allowing for randomly generated service names.